File: tmdb-core/src/main/java/dev/reuss/tmdb/common/genre/Genre.java
package dev.reuss.tmdb.common.genre;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * TMDB genre.
 *
 * @param id   TMDB genre id
 * @param name localized genre name
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record Genre(
        int id,
        String name
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/common/genre/Genre.kt
package dev.reuss.tmdb.common.genre

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * TMDB genre.
 *
 * @property id   TMDB genre id
 * @property name localized genre name
 */
@JvmRecord
data class Genre(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("name")
    val name: String?,
) : TmdbModel
